src/idl_gen_js.cpp

```diff
@@ -524,10 +523,30 @@ class JsGenerator : public BaseGenerator {
     return "NS" + std::to_string(HashFnv1a<uint64_t>(file.c_str()));
   }
 
-  static std::string GenPrefixedImport(const std::string &full_file_name,
-                                       const std::string &base_file_name) {
+  std::string GenPrefixedImport(const std::string &full_file_name,
+                                const std::string &base_name) {
+    // either keep the include path as it was
+    // or use only the base_name + kGeneratedFileNamePostfix
+    std::string path;
+    if (parser_.opts.keep_include_path) {
+      auto it = parser_.included_files_.find(full_file_name);
+      FLATBUFFERS_ASSERT(it != parser_.included_files_.end());
+      path =
+          flatbuffers::StripExtension(it->second) + kGeneratedFileNamePostfix;
+    } else {
+      path = base_name + kGeneratedFileNamePostfix;
+    }
+
+    // add the include prefix
+    path = flatbuffers::ConCatPathFileName(parser_.opts.include_prefix, path);
+
+    if (!flatbuffers::IsAbsPath(path)) {
+      // make it a strictly relative path
+      path = std::string(".") + kPathSeparator + path;
+    }
+
     return "import * as " + GenFileNamespacePrefix(full_file_name) +
-           " from \"./" + base_file_name + "\";\n";
+           " from \"" + path + "\";\n";
   }
 
   // Adds a source-dependent prefix, for of import * statements.
```

Comment:
Did this change have any effect on any of the generated files?

Reply:
Not that I ve seen, I ve seen some error on travis when I did not use "./" for relative paths.
But how can we make this work for windows?

Reply:
Should we only support relative paths in include statements in .fbs files?

Reply:
Thats my hope to, so we leave the behavior as it was! I revert it 👍